cmd/object-api-utils.go

```diff
@@ -163,6 +164,38 @@ func pathJoin(elem ...string) string {
 	return path.Join(elem...) + trailingSlash
 }
 
+// pathClean - is path.Clean() but retains trailing "/" and returns
+// empty string for empty strings not ".".
+func pathClean(p string) string {
+	dotdot := strings.HasPrefix(p, "../")
+	if dotdot {
+		// Make sure we add a separator to
+		// facilitate path.Clean behavior
+		// to kick in.
+		p = slashSeparator + p
+	}
+
+	// Check path ends with separator, save it.
+	// Make sure we do this before path.Clean()
+	// path.Clean() removes trailing separator.
+	sp := hasSuffix(p, slashSeparator)
```

Comment:
We are using `hasSuffix` here and `strings.HasPrefix` above - when a `hasPrefix` that works for windows is available. In either case because there is nothing to lowercase there is no problem, but we can be consistent and use either the std lib funcs or our own windows compatible funcs.